src/channels.ml

```diff
@@ -77,13 +77,47 @@ let of_seekable_fd fd =
     return () in
   return { c with skip }
 
-let sslctx =
-  Ssl.init ();
-  Ssl.create_context Ssl.SSLv23 Ssl.Client_context
+let _ = Ssl.init ()
 
-let of_ssl_fd fd =
+let default_good_ciphers = "!EXPORT:RSA+AES128-SHA256"
+let default_legacy_ciphers = "RSA+AES256-SHA:RSA+AES128-SHA:RSA+RC4-SHA:RSA+DES-CBC3-SHA"
+
+let sslctx_cached = ref None
+
+let gen_sslctx legacy ciphers =
+  let ctx =
+    (* Here SSLv23 means ALL protocols *)
+    Ssl.create_context
+      (if legacy then Ssl.SSLv23 else Ssl.TLSv1_2)
```

Comment:
Why does SSLv23 mean all protocols when passed to Ssl.create_context, but not below where it is passed to Ssl.disable_protocols?   Would it be worthwhile adding a local alias called something like "SSL_all"?